azure-pipelines.yml

```diff
@@ -197,3 +198,68 @@ jobs:
                --data "{\"text\":\"<!here> *FAILED* $(Agent.JobName): <https://dev.azure.com/digitalasset/ghcide/_build/results?buildId=$(Build.BuildId)|$MESSAGE>\n\"}" \
                $(Slack.URL)
         condition: and(failed(), eq(variables['Build.SourceBranchName'], 'master'))
+  - job: ghcide_stack_windows
```

Comment:
Can we use a single matrix for all platforms? It seems a bit weird that the Windows test setup looks very different from the Linux test setup and also makes it hard to figure out what is different. The matrix setup definily looks nicer than the mess we have currently.

Reply:
well the bash script and cache handling for linux and windows are different and i am not sure if they can be integrated without too much if's
what about a matrix per os like in hie?